### PR TITLE
Add libslirp

### PR DESCRIPTION
**Pull request by ivanholmes**
Left untested for gcc2 because I don't have a gcc2 system.

Adding as it is a dependency of melonDS.

### CODE REVIEW

File: net-libs/libslirp/libslirp-4.7.0.recipe [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+REQUIRES_devel="
+	libslirp$secondaryArchSuffix == $portVersion base
+	"
+
+BUILD_REQUIRES="
+	haiku${secondaryArchSuffix}_devel
+	devel:libglib_2.0$secondaryArchSuffix
+	"
+BUILD_PREREQUIRES="
+    cmd:cmake
```
Comment (augiedoggie) [inline net-libs/libslirp/libslirp-4.7.0.recipe:38]: Is cmake actually being used?

Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe]: No, was left in there from the recipe I adapted, I've removed it now :)

Comment (Begasus) [reply net-libs/libslirp/libslirp-4.7.0.recipe]: Haikuporter comes with a few generic recipes, if you add them to the templates you could use them when starting on a new recipe (I do it also, I even created a generic python recipe on the other laptop for this.
generic-recipes

File: net-libs/libslirp/patches/libslirp-4.7.0.patchset [hunk 2]
```diff
@@ -0,0 +1,24 @@
+From 78faa99ec2886bdf8af8e8719980ed9868851ae4 Mon Sep 17 00:00:00 2001
```
Comment (Begasus) [inline net-libs/libslirp/patches/libslirp-4.7.0.patchset:1]: Could you upstream this patch?

Comment (ivanholmes) [reply net-libs/libslirp/patches/libslirp-4.7.0.patchset:1]: https://gitlab.freedesktop.org/slirp/libslirp/-/merge_requests/123
We'll see how this goes

Comment (ivanholmes) [reply net-libs/libslirp/patches/libslirp-4.7.0.patchset:1]: They've already merged it upstream :)

File: net-libs/libslirp/libslirp-4.7.0.recipe [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+CHECKSUM_SHA256="9398f0ec5a581d4e1cd6856b88ae83927e458d643788c3391a39e61b75db3d3b"
+SOURCE_DIR="libslirp-v$portVersion"
+PATCHES="libslirp-$portVersion.patchset"
+
+ARCHITECTURES="all ?x86_gcc2"
+SECONDARY_ARCHITECTURES="?x86_gcc2 x86"
+
+PROVIDES="
+	libslirp$secondaryArchSuffix = $portVersion
+	lib:libslirp$secondaryArchSuffix = 4
```
Comment (augiedoggie) [inline net-libs/libslirp/libslirp-4.7.0.recipe:18]: This should probably use the common libVersion/libVersionCompat variables with libVersion set to 0.4.0.  You can see examples in various recipes and in @Begasus's [wiki page](https://github.com/Begasus/haikuports/wiki/Useful-variables-for-recipes#libversion-and-libversioncompat)

File: net-libs/libslirp/libslirp-4.7.0.recipe [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+HOMEPAGE="https://gitlab.freedesktop.org/slirp/libslirp/"
+COPYRIGHT="1995, 1996 Danny Gasparovski"
+LICENSE="BSD (3-clause)"
+REVISION="1"
+SOURCE_URI="$HOMEPAGE-/archive/v$portVersion/libslirp-v$portVersion.tar.gz"
+CHECKSUM_SHA256="9398f0ec5a581d4e1cd6856b88ae83927e458d643788c3391a39e61b75db3d3b"
+SOURCE_DIR="libslirp-v$portVersion"
+PATCHES="libslirp-$portVersion.patchset"
+
+ARCHITECTURES="all ?x86_gcc2"
```
Comment (augiedoggie) [inline net-libs/libslirp/libslirp-4.7.0.recipe:13]: You can probably just disable x86_gcc2 as a primary and remove it from SECONDARY_ARCHITECTURES.  The port will need patching to make it c++98 compatible.

File: net-libs/libslirp/libslirp-4.7.0.recipe (hunk 3 above)
Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe:18]: Ah yes, good idea. Will do shortly

Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe:21]: Done, thanks

Comment (Begasus) [reply net-libs/libslirp/libslirp-4.7.0.recipe:21]: You need to add $libVersionCompat here (and in the PROVIDES_devel):
```diff
- 	lib:libslirp$secondaryArchSuffix = 4
+ 	lib:libslirp$secondaryArchSuffix = $libVersionCompat
```

File: net-libs/libslirp/libslirp-4.7.0.recipe [hunk 5]
```diff
@@ -0,0 +1,69 @@
...
+	"
+
+defineDebugInfoPackage libslirp$secondaryArchSuffix \
+	"$libDir"/libslirp.so.$libVersion
+
+BUILD()
+{
+	meson build --buildtype=debugoptimized \
+		--prefix=$prefix --includedir=$includeDir \
+		--libdir=$libDir --datadir=$dataDir --bindir=$binDir
```
[the remark this answers is not in the document]
Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe]: Removed, thanks

File: net-libs/libslirp/libslirp-4.7.0.recipe (hunk 3 above)
Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe]: Oops, of course... done now

File: net-libs/libslirp/libslirp-4.7.0.recipe [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+{
+	ninja -C build install
+
+	# prepare development lib links
+	prepareInstalledDevelLib libslirp
+	fixPkgconfig
+
+	# devel package
+	packageEntries devel $developDir
+}
```
[the remark this answers is not in the document]
Comment (ivanholmes) [reply net-libs/libslirp/libslirp-4.7.0.recipe:69]: So there is, the README claims there is not. Added now

**Comment by Begasus** [comment]
PS, I did look at their gitlab source page, think they removed the test not so long ago, but it did still work with this version :)

**Comment by ivanholmes** [comment]
> PS, I did look at their gitlab source page, think they removed the test not so long ago, but it did still work with this version :)

Nice one, I suppose we can remove this part if it doesn't work when the next version comes out.

**Pull request merged**